bitcoin/src/psbt/error.rs

```diff
@@ -90,6 +92,7 @@ impl fmt::Display for Error {
             Error::InvalidKey(ref rkey) => write!(f, "invalid key: {}", rkey),
             Error::InvalidProprietaryKey => write!(f, "non-proprietary key type found when proprietary key was expected"),
             Error::DuplicateKey(ref rkey) => write!(f, "duplicate key: {}", rkey),
+            Error::FutureVersion(ref e) => write_err!(f, "unrecognized PSBT version"; e),
```

Comment:
I realized this will produce silly-looking messages like `unrecognized PSBT version: Unknown future PSBT version 42` but I don't see a clear way how to make them better. A not-very-clear way is to use `fmt::Display::fmt(e, f)` (with a comment that this is intentionally flattened) and then impl source like this:

```rust
Error::FutureVersion(ref e) => e.source(),
```

Effectively flattening the variant. Any ideas?

Reply:
`Error::FutureVersion(ref e) => e.fmt(f)` will produce Unknown future PSBT version 42. How's that? Error handling the right way is a big rust blind spot for me to improve.